patina_dxe_core/src/allocator.rs

```diff
@@ -986,7 +986,7 @@ fn process_hob_allocations(hob_list: &HobList) {
 
         if (stack_address == 0) || (stack_length == 0) {
```

Comment:
if we just turn these into some asserts, We would be able to reduce the "layers" by 1, because we would remove the else statement, which tabs it over by one.

```rust
assert_ne!(stack_address, 0, "Invalid Stack Configuration: stack base address must be non-zero");
assert_ne!(stack_length, 0, "Invalid Stack Configuration: stack length must be non-zero");
```

It would show an error like this:

```
thread 'main' panicked at 'Invalid Stack Configuration: stack base address must be non-zero
assertion failed: `(left != right)`
  left: `0`,
 right: `0`', src/main.rs:10:5
```

Another option (probably reduces binary size, less logging)

```rust
assert!(stack_address != 0 && stack_length != 0,
    "Invalid Stack Configuration: Stack base address {stack_address:#X} for len {stack_length:#X}");
```

Reply:
Good suggestion. I just used a single assert.